rust/connlib/tunnel/src/client.rs

```diff
@@ -70,30 +70,29 @@ where
             debug_assert!(false, "The original message should be a DNS query for us to ever call write_dns_lookup_response");
             return Ok(());
         };
-        let response = match response {
+        let response = match response.map_err(|err| err.kind().clone()) {
             Ok(response) => message.add_answers(response.records().to_vec()),
-            Err(err) => {
-                if let hickory_resolver::error::ResolveErrorKind::NoRecordsFound {
-                    soa,
-                    response_code,
-                    ..
-                } = err.kind()
-                {
-                    if let Some(soa) = soa {
-                        message.add_name_server(soa.clone().into_record_of_rdata());
-                    }
-
-                    message.set_response_code(*response_code)
-                } else {
-                    return Err(err.into());
+            Err(hickory_resolver::error::ResolveErrorKind::NoRecordsFound {
+                soa,
```

Comment:
I think this might be a bug.

Reply:
It's on purpose, we want to forward the `NoRecordsFound` response, otherwise the DNS query needs to timeout before continuing making a noticeable delay in DNS resolution.